File: integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/bugs/DateEntity.java
package io.quarkus.it.mongodb.panache.bugs;

import java.time.Instant;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.Date;

import io.quarkus.mongodb.panache.PanacheMongoEntity;

/**
 * An entity that have all the supported date format.
 * Asserting #6566 and possibility other date issues.
 */
public class DateEntity extends PanacheMongoEntity {
    public Date dateDate;
    public LocalDate localDate;
    public LocalDateTime localDateTime;
    public Instant instant;

    public DateEntity() {
        dateDate = new Date();
        localDate = LocalDate.now();
        localDateTime = LocalDateTime.now();
        instant = Instant.now();
    }
}

File: integration-tests/mongodb-panache-kotlin/src/main/kotlin/io/quarkus/it/mongodb/panache/bugs/DateEntity.kt
package io.quarkus.it.mongodb.panache.bugs

import io.quarkus.mongodb.panache.kotlin.PanacheMongoCompanion
import io.quarkus.mongodb.panache.kotlin.PanacheMongoEntity
import java.time.Instant
import java.time.LocalDate
import java.time.LocalDateTime
import java.util.Date

/**
 * An entity that have all the supported date format. Asserting #6566 and possibility other date
 * issues.
 */
class DateEntity(
    var dateDate: Date = Date(),
    var localDate: LocalDate = LocalDate.now(),
    var localDateTime: LocalDateTime = LocalDateTime.now(),
    var instant: Instant = Instant.now(),
) : PanacheMongoEntity() {
    companion object : PanacheMongoCompanion<DateEntity>
}
